Language links refactor

mdlincoln opened the pull request:
This is a major back-end modification that adjusts how to point to and from translations and their sources. As a result, we should now theoretically be able to host any number of languages on the site. Another nice benefit is that clicking on the `[es]` or `[en]` links at the top of the page should now - when there is a translation / translation source - take you to the companion page for the one you were just on, rather than dumping you all the way back at the home page for that language.

This works by doing two searches every time a page is generated: 

1. if the page has YAML metadata pointing to a source translation, this identifies that exact page and makes that source's metadata available when rendering the translation's page.
2. Searches all other pages across the site to see if they're _translations of the page being rendered_. If so, makes all the metadata of those pages available as needed.

This results in only one change for the editorial guidelines for producing translations: instead of using `redirect-from: /lessons/LESSON-SLUG`, editors will now use `original: LESSON-SLUG` (note no leading directory information). I've already gone through and updated all the existing translations for this. All new translations of non-lesson pages will also require this `original` tag. Doing this means we have a dedicated bit of metadata for finding translation sources, and leaves `redirect-from` to do its important work of ensuring good redirects without saddling it with other responsibilities.

I still have a few open tasks:

- [x] Enforce checking for `original` tags appropriately (might be a separate PR since #816 is also underway right now)
- [x] Made a little "translation map" that is mostly just useful for internal debugging, but it needs some kind of link on the site. [netlify preview](https://deploy-preview-818--ph-dev.netlify.com/translation-concordance)
- [x] render translation map as a table

Closes: #785 

Re: #651

mdlincoln committed:
use snippets for author info heading

mdlincoln committed:
refactor translation<>original links using where_exp

mdlincoln committed:
for citation use translation_date rather than trying to check language

mdlincoln committed:
use direct paths to spanish lessons rather than redirects

mdlincoln committed:
bundle update

mdlincoln committed:
add "original" metadata to all translated lessons

mdlincoln committed:
Functioning page-to-page translation links

Refactors the generation of translation links so that, for any page that
points to an "original" source language, we can identify its
translations and generate the proper links in the menu header

mdlincoln committed:
add "original" links for existing spanish lessons

mdlincoln committed:
only create twitter cards for lessons and posts

mdlincoln committed:
Merge branch 'gh-pages' into language-refactor

mdlincoln committed:
Create autogenerated translation concordance

Useful for debugging

mdlincoln committed:
Describe how to use `original` YAML tag in ES editorial guide

mdlincoln committed:
Document how the new translation links work.

arojascastro reviewed (approved):
Ok, thank you for this.  I understand that you changed already the metadata of the existing translations by replacing `redirect` with `original` and that you also modified our guidelines for editors, right? Great job!

mdlincoln committed:
render translation concordance as table

mdlincoln committed:
add translation concordance to about menu

mdlincoln committed:
enforce "original" links on spanish lessons

This does mean that lessons submitted originally in Spanish will need to
use the "exclude_from_check" tag.
